scripts/yul_coverage.sh

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+
+set -e
+
+ROOT_DIR="$(dirname "$0")"/..
+
+for arg in "$@"; do
+  if [ "$arg" == "--no-stats" ]; then
+    NO_STATS=1
+  elif [ "$arg" == "--no-errors" ]; then
+    SHOW_NO_ERRORS=1
+  elif [ "$arg" == "--internal-compiler-errors" ]; then
+    SHOW_UNIMPLEMENTED_FEATURE_ERRORS=1
+  elif [ "$arg" == "--unimplemented-feature-errors" ]; then
+    SHOW_INTERNAL_COMPILER_ERRORS=1
+  elif [ "$arg" == "--other-errors" ]; then
+    SHOW_OTHER_ERRORS=1
+  elif [ "$arg" == "--list-files" ]; then
+    ONLY_LIST_FILES=1
+  else
+    echo "$(basename "${0}") [--no-stats] [--no-errors] [--internal-compiler-errors] [--unimplemented-feature-errors] [--other-errors] [--list-files]"
+    exit 0
+  fi
+done
+
+show_output_if() {
+  local VAR=${1}
+  if [ -n "${VAR}" ]; then
+    echo "${SOL_FILE}"
+    if [ -z "${ONLY_LIST_FILES}" ]; then
+      echo "${OUTPUT}"
+      echo ""
+    fi
+  fi
+}
+
+FAILED=()
+SUCCESS=()
+SOLC=${SOLC:-solc}
+test_file() {
+  local SOL_FILE
+  local OUTPUT
+  SOL_FILE=${1}
+
+  set +e
+  OUTPUT=$("${SOLC}" --ir "${SOL_FILE}" 2>&1)
+  RESULT=$?
+  set -e
```

Comment:
This unfortunately has an unintended consequence: the script does not fail if `solc` does not exist. It does report `1326 OtherError(s)` but I think that a separate check that the executable exists would make things much clearer for the user of the script.